### ISSUE TITLE
Likes not added to redis

### ISSUE THREAD

**Issue by cvrsor** [schneems/likeable#10]
Hi,

I'm developing a rails3 application using RedisToGo Nano as my development environment and a Mini instance in Heroku for production. I have closely followed the provided Likeable sample application and added the _Likeable::UserMethods_ to my user model as well as the _Likeable module_ to my posts model (and also added the likes_controller.rb, routes.rb, and  like_helper.rb code as appropriate).

Problem is some likes are registered in redis and some are not at all! The bad ones do not create a key at all, meaning, after clicking the _like_ link I run the REDIS.keys command and do not see anything relevant to the specific record.

I used the link tag provided in the _posts index view_ in the sample application. _Every time_ I click the link, the POST request is executed correctly, it includes "resource_name"=>"Post", "resource_id"=>"XX" in the parameters, and returns a 'Success' notice as it should in my view.

My only deviation from the sample application is in the setup of redis, I did: 
in config/environments/development.rb:
```ENV["REDISTOGO_URL"] = 'redis://username:password@myserver.redistogo.com:myport'  

``````

and in config/initializers/redis.rb
```uri = URI.parse(ENV["REDISTOGO_URL"])
REDIS = Redis.new(:host => uri.host, :port => uri.port, :password => uri.password)
``````

Any ideas?
Many thanks for your help,

**Comment by schneems** [comment]
This is an odd one. If you're saying it works sometimes, for some posts, but not other posts. Is there any pattern with the posts that cannot be liked? Perhaps certain id's that cannot be liked?

I would recommend tailing the log to make sure that the initial post works to the controller

`heroku logs --tail`

For some additional insight you can take a look at the result of `target = Likeable.find_by_resource_id(params[:resource_name], params[:resource_id])
`

by calling `puts target.inspect` on the element to ensure it is being properly recovered from the database. 

if that works every time, you can ensure the proper keys are being generated

```
puts target.like_key.inspect
puts current_user.like_key
```

If those look correct it is possible that your connection to redis is intermittent. You can try writing to redis directly

```
redis = Likeable.redis
puts redis.inspect # make sure this matches your redis-to-go information
puts redis.hset(target.like_key, current_user.id, Time.now.to_f)
puts redis.hkeys(target.like_key)
```

Let me know what the result of each of those tests are, and I can help trouble-shoot more. What version of Rails are you on?

**Comment by cvrsor** [comment]
Hi Richard,

Many thanks for your reply and help. Indeed, it is a very strange error: here's what I found after inspecting the elements you suggest:
1. The post reaches the controller, I can see the 'Processing by LikesController#create as HTML' in the terminal window
2. `puts target.inspect` correctly recovers the 'Post' element from database, all fields are shown correctly in the terminal window (_#<Post id: 60_, ...)
3. `puts target.like_key.inspect` correctly returns _"Posts:like:60"_
4. `puts current_user.like_key` returns _Users:like:1_

So I guess everything works fine until here.
Also, running `redis = Likeable.redis` in _rails console_ spits out correctly my RedisToGo connection string, and I can set the hash.

Really strange error..

Only pattern I've noticed is that **the first attempt to like/unlike always succeeds** and every subsequent attempt always fails..And if I restart the server and try the same record it succeeds, it seems that the redis connection works only for the fist POST and then its gone!

Just in case you spot any stupid mistakes:
_config/initializers/likeable.rb_```
Likeable.setup do |likeable|
  uri = URI.parse(ENV["REDISTOGO_URL"])
  likeable.redis  = Redis.new(:host => uri.host, :port => uri.port, :password => uri.password)
end

```

Hey, many thanks for your help, really appreciated!
```

**Comment by cvrsor** [comment]
Don't know if it helps, I'm running Rails v.3.0.10. I really have no idea where to start for debugging..

**Comment by cvrsor** [comment]
Hi Richard,

I think my issue is the same as [issue #11](https://github.com/schneems/Likeable/issues/11) opened by styx!

I had not spotted this before, my 'unsuccessful' posts were sending in the correct resource id, however the _target_ returned by _find_by_resource_id_ is **always** the first post that is made (from when server started).

Something funny could be going on in _likeable / module_methods.rb_ and specifically in `def find_one(klass, id)` 
Thanks!

**Comment by schneems** [comment]
Thanks again for reporting this, i updated notes in #11. Should be fixed with Gem Version 0.1.1
